probe-rs/src/architecture/arm/core/armv7a.rs

```diff
@@ -262,17 +262,18 @@ impl<'probe> CoreInterface for Armv7a<'probe> {
     }
 
     fn halt(&mut self, timeout: Duration) -> Result<CoreInformation, Error> {
-        let address = Dbgdrcr::get_mmio_address(self.base_address);
-        let mut value = Dbgdrcr(0);
-        value.set_hrq(true);
-
-        self.memory.write_word_32(address, value.into())?;
+        if !matches!(self.state.current_state, CoreStatus::Halted(_)) {
```

Comment:
Maybe add a note here that this only works because by default (unknown) we assume that the core is running. Also maybe we should have an unknown state in the future :)

Reply:
See the other comment - we read the current state in `new()` so this is safe in all cases.